shared/js/background/dnr-user-allowlist.js

```diff
@@ -103,10 +103,9 @@ export async function refreshUserAllowlistRules (allowlistedDomains) {
 
 /**
  * Retrieve a normalized and sorted list of user denylisted domains.
- * @returns {Promise<string[]>}
+ * @returns {string[]}
  */
-export async function getDenylistedDomains () {
-    await settings.ready()
```

Comment:
Why stop waiting for settings to be ready, when we're using settings?

Reply:
This function was never called in a context where settings could not be ready - most of the DNR functions don't start running until config or the blocklist is loaded, and those require settings to be ready. Here, removing the `settings.ready` call, allows this function to be sync.